otherlibs/runtime_events/runtime_events_consumer.c

```diff
@@ -88,57 +88,47 @@ struct caml_runtime_events_cursor {
 
 /* C-API for reading from an runtime_events */
 
-runtime_events_error
-caml_runtime_events_create_cursor(const char_os* runtime_events_path, int pid,
-                             struct caml_runtime_events_cursor **cursor_res) {
-  int ret;
-
-#ifndef _WIN32
-  int ring_fd;
-  struct stat tmp_stat;
-#endif
-
-  struct caml_runtime_events_cursor *cursor =
-      caml_stat_alloc_noexc(sizeof(struct caml_runtime_events_cursor));
+/** Return a new string with the path of the ring file */
+static runtime_events_error format_runtime_events_loc(
```

Comment:
Note that (because C is antediluvian) enumeration constants such as `E_PATH_FAILURE` are of type `int`, so here you are forcing an implicit conversion into the enum type, and then at the call site there's an implicit conversion back to `int`. So maybe this return type should be `int`? For this reason, I never name enum types (so I never have variables or slots of enum types).